python_modules/libraries/dagster-aws/dagster_aws/ecs/launcher.py

```diff
@@ -344,3 +351,37 @@ def _reuse_task_definition(self, task_definition, metadata, image, secrets):
 
     def _task_metadata(self):
         return default_ecs_task_metadata(self.ec2, self.ecs)
+
+    @property
+    def supports_check_run_worker_health(self):
+        return True
+
+    def check_run_worker_health(self, run: PipelineRun):
+
+        tags = self._get_run_tags(run.run_id)
+
+        if not (tags.arn and tags.cluster):
+            return CheckRunHealthResult(WorkerStatus.UNKNOWN, "")
+
+        tasks = self.ecs.describe_tasks(tasks=[tags.arn], cluster=tags.cluster).get("tasks")
+        if not tasks:
+            return CheckRunHealthResult(WorkerStatus.UNKNOWN, "")
+
+        t = tasks[0]
+
+        running_statuses = ["PROVISIONING", "PENDING", "ACTIVATING", "RUNNING"]
+        stopped_statuses = ["DEACTIVATING", "STOPPING", "DEPROVISIONING", "STOPPED"]
+
+        if t.get("lastStatus") in running_statuses:
+            return CheckRunHealthResult(WorkerStatus.RUNNING)
+        elif t.get("lastStatus") in stopped_statuses:
+            for c in t.get("containers"):
```

Comment:
not sure if this is the best way to check if the task failed, but having issues finding another approach to getting that info. could parse logs, but would maybe prefer to avoid that if i can